lib/cppcheck.cpp

```diff
@@ -590,6 +590,112 @@ void CppCheck::checkSimplifiedTokens(const Tokenizer &tokenizer)
         executeRules("simple", tokenizer);
 }
 
+#ifdef HAVE_RULES
+
+static std::string pcreErrorCodeToString(const int pcreExecRet)
+{
+    std::string errorMessage;
+    switch (pcreExecRet) {
+    case PCRE_ERROR_NULL:
+        errorMessage = "PCRE_ERROR_NULL";
```

Comment:
You could return the string directly. And returning const char* instead of std::string might be more efficient.